CHANGELOG.md

```diff
@@ -1,30 +1,32 @@
 # Release History
 
-## 1.29.0 (TBD)
+## 1.29.0 (2025-03-05)
 
 ### Snowpark Python API Updates
 
 #### New Features
 
-- Added support for the following AI-powered functions in `functions.py` (private preview):
+- Added support for the following AI-powered functions in `functions.py` (Private Preview):
   - `ai_filter`
   - `ai_agg`
   - `summarize_agg`
-- Added support for the new FILE SQL type support, with the following related functions in `functions.py` (private preview):
-  - fl_get_content_type
-  - fl_get_etag
-  - fl_get_file_type
-  - fl_get_last_modified
-  - fl_get_relative_path
-  - fl_get_scoped_file_url
-  - fl_get_size
-  - fl_get_stage
-  - fl_get_stage_file_url
-  - fl_is_audio
-  - fl_is_compressed
-  - fl_is_document
-  - fl_is_image
-  - fl_is_video
+- Added support for the new FILE SQL type support, with the following related functions in `functions.py` (Private Preview):
+  - `fl_get_content_type`
+  - `fl_get_etag`
+  - `fl_get_file_type`
+  - `fl_get_last_modified`
+  - `fl_get_relative_path`
+  - `fl_get_scoped_file_url`
+  - `fl_get_size`
+  - `fl_get_stage`
+  - `fl_get_stage_file_url`
+  - `fl_is_audio`
+  - `fl_is_compressed`
+  - `fl_is_document`
+  - `fl_is_image`
+  - `fl_is_video`
+- Added support for importing third-party packages from PyPi using Artifact Repository (Private Preview):
+  - Use keyword arguments `artifact_repository` and `artifact_repository_packages` to specify your `ARTIFACT_REPOSITORY` and `ARTIFACT_REPOSITORY_PACKAGES` respectively when registering stored procedures or user defined functions.
```

Comment:
I suggest 
1. we make it explicit which APIs accept the two kwargs.
2. Mark it Private Preview.
3. Use words instead of `ARTIFACT_REPOSITORY` and `ARTIFACT_REPOSITORY_PACKAGES`